tests/ClientTest.php

```diff
@@ -336,6 +338,68 @@ public function sampleRateAbsoluteDataProvider(): array
         ];
     }
 
+    public function testProcessDiscardsEventWhenItIsSampledDueToSampleRateOption(): void
```

Comment:
Test name not clear, it seems to me nearly the same meaning of line 316

Reply:
I've merged together the two tests and improved their code in the process 👍